src/zenml/zen_stores/schemas/component_schemas.py

```diff
@@ -65,6 +68,10 @@ class StackComponentSchema(SQLModel, table=True):
     )
     user: "UserSchema" = Relationship(back_populates="components")
 
+    artifacts: List["ArtifactSchema"] = Relationship(
```

Comment:
Yes, only the artifact store will have this. However, these SQLModel relationships are only helper attributes that make it easier to get one schema from another, they don't affect the underlying DB in any way, so we are not actually coding this into the `stack_component` table, rather it just adds a `stack_component.artifacts` attribute in Python that you can use instead of querying that information

Reply:
Yeah but exactly that might be the confusing part right, that `stack_component.artifacts` exists but doesn't return anything useful. E.g. when having an orchestrator component model, just seeing this attribute would make me believe that I could get all the artifacts for all runs using this orchestrator. In any case, probably something we can do in the "final" model rework later

Reply:
Yeah but exactly that might be the confusing part right, that `stack_component.artifacts` exists but doesn't return anything useful. E.g. when having an orchestrator component model, just seeing this attribute would make me believe that I could get all the artifacts for all runs using this orchestrator. In any case, probably something we can do in the "final" model rework later

Reply:
Noted in down in the 1500 ticket 👍

Reply:
I agree with Michael on this one. This can get a bit confusing. I am not fully through the PR just yet but is there a place where we are using this functionality already?